library/cool/invariance.red

```diff
@@ -41,3 +41,85 @@ def swap-fun/path (A : type) : (f : bool → A) → path _ (swap-fun A (swap-fun
     fun→pair/path A j → fun→pair/path A j
   in
   (f : fun→pair/path A i) → path (fun→pair/path A i) (swapcoe (swapcoe f)) f
+
+-- A cooler example?
+
+import data.list
+import data.nat
+import data.unit
+import paths.list
+import basics.isotoequiv
+
+def nat→list : nat → list unit =
+  elim [
+  | zero → nil
+  | suc (_ → xs) → cons triv xs
+  ]
+
+def length (A : type) : list A → nat =
+  elim [
+  | nil → zero
+  | cons _ (_ → n) → suc n
+  ]
+
+-- TODO ??
```

Comment:
what does "TODO" mean here?

Reply:
I just wrote it to mean this code might not even be useful? Don't worry about it.